ARM: dts: Add support for Olimex AM3352-SOM

vinifr opened the pull request:
Add a dts file for Olimex AM3352-SOM board. The board does not use the PMIC
tps65217 and does not have many peripherals present in beaglebone. Thus, a
specific dtsi file (am335x-som-common.dtsi) is needed.

Signed-off-by: Dimitar Gamishev <hehopmajieh@debian.bg>
Signed-off-by: Vinicius Maciel <viniciusfre@gmail.com>

RobertCNelson commented:
Something wrong? I was going to gueue this up on Monday? Let me know..

vinifr commented:
Something is wrong. The two Ethernet interfaces are not detected. Just working with https://github.com/OLIMEX/linux-3.12.10-ti2013.12.01-am3352_som/blob/master/drivers/net/ethernet/ti/cpsw.c

My change seems it's not ok to your cpsw.c:

&mac {
	pinctrl-names = "default", "sleep";
	pinctrl-0 = <&cpsw_default>;
	pinctrl-1 = <&cpsw_sleep>;
	slaves = <2>;
	dual_emac = <1>;
	
	cpsw_emac0 {
		dual_emac_res_vlan = <1>;
	};

	cpsw_emac1 {
		dual_emac_res_vlan = <2>;
	};
};

&cpsw_emac0 {
	phy_id = <&davinci_mdio>, <0>;
	phy-mode = "mii";
};

&cpsw_emac1 {
	phy_id = <&davinci_mdio>, <1>;
	phy-mode = "mii";
};

RobertCNelson commented:
I think your tripping on linux/commit/e6c31b3467ac383ba57ae41146d31f0ef29da3c3

Drop the am335-bone compat..

RobertCNelson commented:
https://github.com/beagleboard/linux/commit/e6c31b3467ac383ba57ae41146d31f0ef29da3c3

vinifr commented:
If I understood you correctly, removing "ti,am335x-bone" from am335x-olimex-som.dts still does not work. It's very strange, There is not even a mention to davinci_mdio.c functions or cpsw.c: http://pastebin.com/aDPBYXC7 ... I setup kernel log level to 7.

vinifr commented:
Hi RobertCNelson, 
Finally the ethernet interfaces worked! A simple status = "okay" solved the problem. 

davinci_mdio: dt: updated phy_id[0] from phy_mask[fffffffc]
davinci_mdio: dt: updated phy_id[1] from phy_mask[fffffffc]

RobertCNelson commented:
That's great it worked, but i see the phy_mask bug so only one probally works, i'll edit out the "ti,am335x-bone" in the patch..

vinifr commented:
OK. Anyway, I tested both interfaces and both pick up IP.

# ifconfig 
eth2      Link encap:Ethernet  HWaddr 1c:ba:8c:9c:92:49  
          inet addr:192.168.25.173  Bcast:192.168.25.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:6953 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3953 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:9849789 (9.3 MiB)  TX bytes:276379 (269.9 KiB)
          Interrupt:171 

eth3      Link encap:Ethernet  HWaddr 1c:ba:8c:9c:92:4b  
          inet addr:192.168.25.172  Bcast:192.168.25.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:44 errors:0 dropped:0 overruns:0 frame:0
          TX packets:21 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:8655 (8.4 KiB)  TX bytes:3283 (3.2 KiB)

RobertCNelson commented:
That's crazy!

With the patch we get the first phy_mask:

https://github.com/RobertCNelson/ti-linux-kernel-dev/blob/ti-linux-4.1.y/patches/beaglebone/phy/0001-cpsw-search-for-phy.patch#L45

and then we update the both ethernet slaves with the exact same phy_mask. ;)

I'm just really surprised you got two ip's, i was kinda expecting them to clone..

RobertCNelson commented:
Okay, pushed to "4.1.x-ti" and "4.1.x-rt-ti" it'll be part of r36, pushed out on Thursday, noon central.

Thanks!!!

vinifr commented:
Great! Thanks.
